pkg/kubelet/container_bridge.go

```diff
@@ -68,54 +119,15 @@ func createCBR0(wantCIDR *net.IPNet, babysitDaemons bool) error {
 			glog.Error(err)
 		}
 	}
-	glog.V(2).Info("Recreated cbr0 and restarted docker")
+	glog.V(2).Info("Recreated %s and restarted docker", br.Name)
 	return nil
 }
 
-func ensureCbr0(wantCIDR *net.IPNet, promiscuous, babysitDaemons bool) error {
-	exists, err := cbr0Exists()
-	if err != nil {
-		return err
-	}
-	if !exists {
-		glog.V(2).Infof("CBR0 doesn't exist, attempting to create it with range: %s", wantCIDR)
-		return createCBR0(wantCIDR, babysitDaemons)
-	}
-	if !cbr0CidrCorrect(wantCIDR) {
-		glog.V(2).Infof("Attempting to recreate cbr0 with address range: %s", wantCIDR)
-
-		// delete cbr0
-		if err := exec.Command("ip", "link", "set", "dev", "cbr0", "down").Run(); err != nil {
-			glog.Error(err)
-			return err
-		}
-		if err := exec.Command("brctl", "delbr", "cbr0").Run(); err != nil {
-			glog.Error(err)
-			return err
-		}
-		if err := createCBR0(wantCIDR, babysitDaemons); err != nil {
-			glog.Error(err)
-			return err
-		}
-	}
-	// Put the container bridge into promiscuous mode to force it to accept hairpin packets.
-	// TODO: Remove this once the kernel bug (#20096) is fixed.
-	if promiscuous {
-		// Checking if the bridge is in promiscuous mode is as expensive and more brittle than
-		// simply setting the flag every time.
-		if err := exec.Command("ip", "link", "set", "cbr0", "promisc", "on").Run(); err != nil {
-			glog.Error(err)
-			return err
-		}
-	}
-	return nil
-}
-
-// Check if cbr0 network interface is configured or not, and take action
+// Check if the bridge network interface is configured or not, and take action
 // when the configuration is missing on the node, and propagate the rest
 // error to kubelet to handle.
-func cbr0Exists() (bool, error) {
-	if _, err := os.Stat("/sys/class/net/cbr0"); err != nil {
+func bridgeExists(brName string) (bool, error) {
```

Comment:
While you're here, how about just doing:

_, err := netlink.LinkByName(brName)
if os.IsNotExist(err) {
    return false, nil
}